pennylane/ops/op_math/decompositions/unitary_decompositions.py

```diff
@@ -760,3 +761,123 @@ def two_qubit_decomp_rule(U, wires, **__):
     )(U, wires, initial_phase)
     total_phase = initial_phase + additional_phase
     ops.GlobalPhase(-total_phase)
+
+
+def _compute_udv(a, b):
+    r"""Given the matrices `a` and `b`, calculates the matrices `u`, `d` and `v`
+    of Eq. 36 in [arXiv-quant-ph:0504100](https://arxiv.org/pdf/quant-ph/0504100):
+
+    .. math::
+
+        a = u d v \\
+        b = u d^{\dagger} v.
+    """
+
+    # Calculates u and d diagonalizing ab^dagger (Eq.39)
+    ab_dagger = a @ math.conj(b.T)
+    d_square, u = math.linalg.eig(ab_dagger)
+    u, _ = math.linalg.qr(u)
+
+    # complex square root of eigenvalues
+    d = math.exp(1j * math.angle(d_square) / 2)
+
+    # Calculates v using Eq.40
+    v = math.conj(math.diag(d).T) @ math.conj(u.T) @ a
+
+    return u, d, v
+
+
+def multi_qubit_decomposition(U, wires):
+    r"""Decompose an n-qubit unitary :math:`U` (with n > 1) into four (n-1)-qubit unitaries and
```

Comment:
For consistency with one_ and two_qubit_decomposition, you might consider this:

```suggestion
    r"""Decompose a multi-qubit unitary :math:`U` in terms of elementary operations.
```